<?xml version="1.0"?>

<launch>
  <!-- 0-raw, 1-pos, 2-trj -->
  <param name="flight_mode" value="1" type="int" />
  <param name="joy_num" value="1" type="int" />
  <arg name="joy_dev0" default="/dev/input/js0" />
  <arg name="g_vehicle_num" default="1" />
  
  <arg name="tf_prefix" default="" />
  <arg name="roll_trim" default="0" />
  <arg name="pitch_trim" default="0" />
  <arg name="enable_logging" default="True" />
  <arg name="enable_logging_imu" default="True" />
  <arg name="use_ros_time" default="True" />
  <arg name="enable_logging_att" default="True" />
  <param name="enable_logging" value="$(arg enable_logging)" />
  <param name="enable_logging_imu" value="$(arg enable_logging_imu)" />
  <param name="use_ros_time" value="$(arg use_ros_time)" />
  <param name="enable_logging_att" value="$(arg enable_logging_att)" />


  <node pkg="easyfly" type="swarm_Server" name="swarm_Server" output="screen"/>
  <!--node pkg="easyfly" type="swarm_Server_Decentralized" name="swarm_Server_Decentralized" output="screen"/-->
  <!--node pkg="easyfly" type="swarm_Server_pseudoDecentralized" name="swarm_Server" output="screen"/-->

  <group ns="joygroup0">
    <node pkg="joy" name="joy_node0" type="joy_node" >
      <param name="dev" value="$(arg joy_dev0)" />
    </node>
  </group>

  <group ns="vehicle0">
      <arg name="uri" default="radio://0/80/250K/0xE7E7E7E7E7"/>
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
        <param name="uri" value="$(arg uri)" type="string" />
        <param name="group_index" value="0"/>
        <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
        <param name="tf_prefix" value="$(arg tf_prefix)" />
        <param name="roll_trim" value="$(arg roll_trim)" />
        <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test0" output="screen" >
        <param name="group_index" value="0"/>
      </node>
  </group>


  <group ns="vehicle1">
      <arg name="uri" default="radio://0/80/250K/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="1"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test1" output="screen" >
        <param name="group_index" value="1"/>
      </node>
  </group>

  <group ns="vehicle2">
      <arg name="uri" default="radio://0/80/2M/0xE9E9E9E9E9"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="2"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test1" output="screen" >
        <param name="group_index" value="2"/>
      </node>
  </group>

  <!--group ns="vehicle3">
      <arg name="uri" default="radio://0/20/1M/0xE7E7E7E7E7"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="0"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller3" output="screen" >
        <param name="group_index" value="0"/>
      </node>
  </group-->

  <!--group ns="vehicle4">
      <arg name="uri" default="radio://2/60/250K/0xE7E7E7E7E7"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="2"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller4" output="screen" >
        <param name="group_index" value="2"/>
      </node>
  </group-->

  <!--group ns="vehicle5">
      <arg name="uri" default="radio://0/100/1M/0xE7E7E7E7E7"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="0"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller5" output="screen" >
        <param name="group_index" value="2"/>
      </node>
  </group-->

  <!--group ns="vehicle6">
      <arg name="uri" default="radio://4/120/250K/0xE7E7E7E7E7"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="6"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller6" output="screen" >
        <param name="group_index" value="6"/>
      </node>
  </group-->

  <!--group ns="vehicle7">
      <arg name="uri" default="radio://3/20/250K/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="3"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller7" output="screen" >
        <param name="group_index" value="3"/>
      </node>
  </group-->

  <!--group ns="vehicle8">
      <arg name="uri" default="radio://3/40/1M/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="3"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node-->
      <!--node pkg="easyfly" type="swarm_controller" name="swarm_controller8" output="screen" >
        <param name="group_index" value="3"/>
      </node-->
  <!--/group-->

  <!--group ns="vehicle9">
      <arg name="uri" default="radio://2/60/250K/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="2"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller9" output="screen" >
        <param name="group_index" value="2"/>
      </node>
  </group-->

  <!--group ns="vehicle10">
      <arg name="uri" default="radio://1/80/1M/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="2"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test10" output="screen" >
        <param name="group_index" value="2"/>
      </node>
  </group>

  <group ns="vehicle11">
      <arg name="uri" default="radio://0/100/1M/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="3"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test" output="screen" >
        <param name="group_index" value="3"/>
      </node>
  </group-->

  <!--group ns="vehicle12">
      <arg name="uri" default="radio://0/120/250K/0xE8E8E8E8E8"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="12"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="swarm_controller" name="swarm_controller12" output="screen" >
        <param name="group_index" value="12"/>
      </node>
  </group-->

  <!--group ns="vehicle13">
      <arg name="uri" default="radio://3/0/1M/0xE9E9E9E9E9"/>
      
      <node pkg="easyfly" type="swarm_addCf" name="swarm_addCf0" output="screen">
          <param name="group_index" value="3"/>
          <param name="g_vehicle_num" value="$(arg g_vehicle_num)" type="int" />
          <param name="uri" value="$(arg uri)" type="string"/>
          <param name="tf_prefix" value="$(arg tf_prefix)" />
          <param name="roll_trim" value="$(arg roll_trim)" />
          <param name="pitch_trim" value="$(arg pitch_trim)" />
      </node>
      <node pkg="easyfly" type="communication_test" name="communication_test13" output="screen" >
        <param name="group_index" value="3"/>
      </node>
  </group-->

  

<!--node name="learning_Cfs" pkg="easyfly" type="State_learning.py" output="screen"/-->

<node pkg="easyfly" name="commander_test" type="commander_test" output="screen"/>
<!--node pkg="easyfly" type="swarm_linker" name="swarm_linker" output="screen" /-->


</launch>
